src/lava/proc/lif/models.py

```diff
@@ -145,9 +157,151 @@ def run_spk(self):
+        """Spiking activation function for LIF"""
+        return self.v >= self.vth
+
+
+@implements(proc=TernaryLIF, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('floating_pt')
+class PyTernLifModelFloat(AbstractPyLifModelFloat):
+    """Implementation of Ternary Leaky-Integrate-and-Fire neural process in
+    floating point precision. This ProcessModel builds upon the floating
+    point ProcessModel for LIF by adding upper and lower threshold voltages.
+    """
+    # Spikes now become 2-bit signed floating point numbers
+    s_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, float, precision=2)
+    vth_hi: float = LavaPyType(float, float)
+    vth_lo: float = LavaPyType(float, float)
+
+    def spiking_activation(self):
+        """Spiking activation for T-LIF: -1 spikes below lower threshold,
+        +1 spikes above upper threshold"""
+        return (-1) * (self.v <= self.vth_lo) + (self.v >= self.vth_hi)
+
+    def reset_voltage(self, spike_vector: np.ndarray):
+        """Reset voltage of all spiking neurons to 0"""
+        self.v[spike_vector != 0] = 0  # Reset voltage to 0 wherever we spiked
+
+
+@implements(proc=LIF, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('bit_accurate_loihi', 'fixed_pt')
+class PyLifModelBitAcc(AbstractPyLifModelFixed):
+    """Implementation of Leaky-Integrate-and-Fire neural process bit-accurate
+    with Loihi's hardware LIF dynamics, which means, it mimics Loihi
+    behaviour bit-by-bit.
+
+    Currently missing features (compared to Loihi 1 hardware):
+        - refractory period after spiking
+        - axonal delays
+
+    Precisions of state variables
+    -----------------------------
+    du: unsigned 12-bit integer (0 to 4095)
+    dv: unsigned 12-bit integer (0 to 4095)
+    bias: signed 13-bit integer (-4096 to 4095). Mantissa part of neuron bias.
+    bias_exp: unsigned 3-bit integer (0 to 7). Exponent part of neuron bias.
+    vth: unsigned 17-bit integer (0 to 131071)
+    """
+    s_out: PyOutPort = LavaPyType(PyOutPort.VEC_DENSE, bool, precision=1)
+    vth: int = LavaPyType(int, np.int32, precision=17)
+
+    def __init__(self):
+        super(PyLifModelBitAcc, self).__init__()
+        self.effective_vth = 0
+
+    def scale_threshold(self):
+        """Scale threshold according to the way Loihi hardware scales it. In
+        Loihi hardware, threshold is left-shifted by 6-bits to MSB-align it
+        with other state variables of higher precision.
+        """
+        self.effective_vth = np.left_shift(self.vth, self.vth_shift)
+        self.isthrscaled = True
+
+    def spiking_activation(self):
+        """Spike when voltage exceeds threshold"""
+        return self.v >= self.effective_vth
+
+
+@implements(proc=TernaryLIF, protocol=LoihiProtocol)
+@requires(CPU)
+@tag('fixed_pt')
+class PyTernLifModelFixed(AbstractPyLifModelFixed):
+    """Implementation of Ternary Leaky-Integrate-and-Fire neural process
+    with fixed point precision. Some state variables are bit-accurate
```

Comment:
This "Some state variables" is too ambiguous, if possible, specify which variables are and which are not bit-accurate with respect to Loihi's architecture.